pkg/controller/buildrun/credentials.go

```diff
@@ -42,7 +46,8 @@ func updateServiceAccountIfSecretNotLinked(sourceSecret *corev1.LocalObjectRefer
 		serviceAccount.Secrets = append(serviceAccount.Secrets, corev1.ObjectReference{
 			Name: sourceSecret.Name,
 		})
+		return true
 	}
 
-	return serviceAccount
```

Comment:
You changed the return value from `*corev1.ServiceAccount` to `bool`,

Can you ensure the service account includes all new secrets which update/append in  func `updateServiceAccountIfSecretNotLinked ` when executing the `r.client.Update(context.TODO(), serviceAccount)` ?

Reply:
This is verified in the unit test here: https://github.com/SaschaSchwarze0/build/blob/sascha-203-sa-reuse/pkg/controller/buildrun/credentials_test.go#L28-L59